corehq/apps/locations/models.py

```diff
@@ -588,6 +581,37 @@ def unarchive(self):
         for loc in [self] + self.descendants:
             loc._unarchive_single_location()
 
+    def _close_case_and_remove_users(self):
+        """
+        Closes linked supply point cases for a location and unassigns the users
+        assigned to that location.
+
+        Used by both archive and delete methods
+        """
+
+        sp = self.linked_supply_point()
+        # sanity check that the supply point exists and is still open.
+        # this is important because if you archive a child, then try
+        # to archive the parent, we don't want to try to close again
+        if sp and not sp.closed:
+            close_case(sp.case_id, self.domain, COMMTRACK_USERNAME)
+
+        _unassign_users_from_location(self.domain, self._id)
+
+    def full_delete(self):
+        """
+        Delete a location and its dependants.
```

Comment:
self.descendants should get all levels of children, and each of the locations returned from that have their couch doc deleted. the sql deletes automatically cascade down when a child has a foreign key relationship with a deleted parent, which is why there is only one sql delete call.